filesystem.go

```diff
@@ -39,24 +39,36 @@ func (f *filesystem) UUID() string {
 func filesystem2_0(source map[string]interface{}) (*filesystem, error) {
 	fields := schema.Fields{
 		"fstype":      schema.String(),
-		"mount_point": schema.String(),
-		"label":       schema.String(),
+		"mount_point": schema.OneOf(schema.Nil(""), schema.String()),
+		"label":       schema.OneOf(schema.Nil(""), schema.String()),
 		"uuid":        schema.String(),
-		// TODO: mount_options when we know the type.
+		// TODO: mount_options when we know the type (note it can be
+		// nil).
 	}
-	checker := schema.FieldMap(fields, nil)
+	defaults := schema.Defaults{
+		"mount_point": "",
+		"label":       "",
+	}
+	checker := schema.FieldMap(fields, defaults)
 	coerced, err := checker.Coerce(source, nil)
 	if err != nil {
 		return nil, WrapWithDeserializationError(err, "filesystem 2.0 schema check failed")
 	}
 	valid := coerced.(map[string]interface{})
 	// From here we know that the map returned from the schema coercion
 	// contains fields of the right type.
-
+	mount_point, ok := valid["mount_point"].(string)
```

Comment:
no need for the if !ok check, as when not present mount_point will be "".

mount_point, _ := valid["mount_point"].(string)